### ISSUE TITLE
Level cap merging doesn't work

### ISSUE THREAD

**Issue by Hy60koshk** [larsiusprime/tdrpg-bugs#1341]
I have a mod that changes level cap. In test_public (pre-2.2.1) it works when I rewrite the defender.xml, but when I try to merge xml's, it does not.
The code in "/_merge/xml/defender.xml":
```
<?xml version="1.0" encoding="utf-8" ?>
<data>
	<level_cap value="200">
		<merge key="value" value="60"/>
	</level_cap>
</data>
```
Also tried `<merge key="value"/>` and `<merge/>`, as you suggested.

Mod files: [200 lvl max.zip](https://github.com/larsiusprime/tdrpg-bugs/files/1604381/200.lvl.max.zip)

**Comment by ManDude** [comment]
Is it just level cap merging that's broken and nothing else?

**Comment by Hy60koshk** [comment]
Not sure, I just submit whatever is broken or seems to be.

**Comment by larsiusprime** [comment]
Whew, that was a tough bug to fix!

It's fixed now in DQDX 2.2.2 1/10 build.

![Uploading 2018-01-10 16_52_49-DefendersQuestI.png…]()

This syntax works:
```
<?xml version="1.0" encoding="utf-8" ?>
<data>
	<level_cap value="200">
		<merge key="value" value="60"/>
	</level_cap>
</data>
```
This will fail (as expected -- because the keyed value doesn't match):
```
<?xml version="1.0" encoding="utf-8" ?>
<data>
	<level_cap value="200">
		<merge key="value" value="99"/>
	</level_cap>
</data>
```

This will fail (because the keyed value is interpreted as "" and thus doesn't match):
```
<?xml version="1.0" encoding="utf-8" ?>
<data>
	<level_cap value="200">
		<merge key="value"/>
	</level_cap>
</data>
```

This will work (because there's no keyed value, it will match with the first tag of the same name at the same level of hierarchy):

```
<?xml version="1.0" encoding="utf-8" ?>
<data>
	<level_cap value="200">
		<merge/>
	</level_cap>
</data>
```

This will not work (because you haven't provided any merge instructions for the payload):

```
<?xml version="1.0" encoding="utf-8" ?>
<data>
	<level_cap value="200">
	</level_cap>
</data>
```

**Comment by larsiusprime** [comment]
The fix for this is up on test_public (windows/non-GL only for now), if you want to test it.
